geowave-vector/src/main/java/mil/nga/giat/geowave/vector/util/FeatureDataUtils.java

```diff
@@ -0,0 +1,79 @@
+package mil.nga.giat.geowave.vector.util;
+
+import mil.nga.giat.geowave.vector.plugin.GeoWaveGTDataStore;
+
+import org.apache.log4j.Logger;
+import org.geotools.feature.simple.SimpleFeatureBuilder;
+import org.geotools.geometry.jts.JTS;
+import org.geotools.referencing.CRS;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.geometry.MismatchedDimensionException;
+import org.opengis.referencing.FactoryException;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+import org.opengis.referencing.operation.MathTransform;
+import org.opengis.referencing.operation.TransformException;
+
+import com.vividsolutions.jts.geom.Geometry;
+
+public class FeatureDataUtils
+{
+	private final static Logger LOGGER = Logger.getLogger(FeatureDataUtils.class);
+
+	public static SimpleFeature defaultCRSTransform(
+			final SimpleFeature entry,
+			final SimpleFeatureType persistedType,
+			final SimpleFeatureType reprojectedType,
+			final MathTransform transform ) {
+
+		// if the feature is in a different coordinate reference system than
+		// EPSG:4326, transform the geometry
+		final CoordinateReferenceSystem crs = entry.getFeatureType().getCoordinateReferenceSystem();
+		SimpleFeature defaultCRSEntry = entry;
+
+		if (!GeoWaveGTDataStore.DEFAULT_CRS.equals(crs)) {
+			MathTransform featureTransform = null;
+			if ((persistedType.getCoordinateReferenceSystem() != null) && persistedType.getCoordinateReferenceSystem().equals(
+					crs) && (transform != null)) {
+				// we can use the transform we have already calculated for this
+				// feature
+				featureTransform = transform;
+			}
+			else if (crs != null) {
+				// this feature differs from the persisted type in CRS,
+				// calculate the transform
+				try {
+					featureTransform = CRS.findMathTransform(
+							crs,
+							GeoWaveGTDataStore.DEFAULT_CRS,
+							true);
+				}
+				catch (final FactoryException e) {
+					LOGGER.warn(
```

Comment:
I know you were re-factoring existing code, but in the case of not being able to transform to EPSG:4326 do we really want to return an feature in the incorrect crs (presumably to be uploaded)?  (Caveat:  I didn't check upstream of this code to see if there's a CRS check)

I would think it would be better not to upload and throw an error in the log (or maybe even runtime exception, though leaning away from that).   Probably last thing we want is to upload inconsistent info.

Or I could be crazy, thoughts?